docs/source/manual/jooq.rst

```diff
@@ -112,9 +112,85 @@ Below is an example of configuring the required binding uses a C3P0 pooling data
+
+Nested Transactions
+"""""""""""""""""""
+
+Actions registered via :code:`whenCommitted` in a nested transaction will only be called after the outermost transaction is committed, and will not be called if a rollback to any outer savepoint occurs during the transaction.
+
+.. code-block:: java
+
+  @JooqTransactional
+  public void outer() {
+    hooks.whenCommitted(() -> foo());
+    inner();
+  }
+
+  @JooqTransactional {
+  public void inner {
+    hooks.whenCommitted(() -> bar());
+  }
+
+  // foo() and then bar() will be called when the outer transaction is committed.
+
+.. code-block:: java
+
+  @JooqTransactional
+  public void outer() {
+    hooks.whenCommitted(() -> foo());
+    try {
+      inner();
+    } catch(Exception e) {
+      // Log or handle exception, but since we caught the inner exception the outer transaction will not be rolled back.
+    }
+  }
+
+  @JooqTransactional {
+  public void inner {
+    hooks.whenCommitted(() -> bar());
+    throw new RuntimeException();  // Raising an exception will cause this inner transaction to rollback.
+  }
+
+  // foo() will be called, but not bar()
+
+Order of Execution
+""""""""""""""""""
+
+Actions registered with :code:`whenCommitted` for a given transaction are executed in the ordered they were registered.
+
+Exception Handling
+""""""""""""""""""
+
+If an action registered with :code:`whenCommitted` for a given transactions throws an exception, then no later registered actions in the same transaction will be called. Since the callback actions are executed *after* a successful commit, an exception in a callback will not cause the transaction to roll back.
+
+Why no rollback hooks?
+""""""""""""""""""""""
+
+Rollback hooks are more difficult to implement robustly. A variety of things can cause an implicit rollback. For instance, if your process is killed, in the middle of a transaction, without a graceful shutdown, your rollback hooks would not run.
+
+Instead of using a rollback hook, try to invert what you are trying to do. Instead of trying to undo something when a transaction fails, instead use a commit hook to delay doing something until after the transaction succeeds.
+
 .. _DataSource: https://docs.oracle.com/javase/8/docs/api/javax/sql/DataSource.html
 .. _DSLContext: https://www.jooq.org/javadoc/latest/org/jooq/DSLContext.html
 .. _DSLContextProvider: https://github.com/cerner/beadledom/tree/master/jooq/src/main/java/com/cerner/beadledom/jooq/DSLContextProvider.java
 .. _JooqTransactional: https://github.com/cerner/beadledom/tree/master/jooq/src/main/java/com/cerner/beadledom/jooq/JooqTransactional.java
 .. _SQLDialect: https://www.jooq.org/javadoc/latest/org/jooq/SQLDialect.html
 .. _ThreadLocalJooqModule: https://github.com/cerner/beadledom/tree/master/jooq/src/main/java/com/cerner/beadledom/jooq/ThreadLocalJooqModule.java
+.. _JooqTransactionalHooks:https://github.com/cerner/beadledom/tree/master/jooq/src/main/java/com/cerner/beadledom/jooq/JooqTransactionalHooqs.java
```

Comment:
Do you need a space after the first : ?

Reply:
Probably, maybe that is why the underscore is rendering weird rather than what I mentioned above.

Reply:
Yes that fixed the link rendering: https://github.com/cerner/beadledom/pull/140/commits/03a822b2d469696fbcb6e89a78fd519211748b5f